### PR TITLE
Add PHPSPEC v6 support.

### PR DESCRIPTION
**Pull request by ptondereau**
Just added 6.0 support in composer.json

### CODE REVIEW

File: spec/Exception/NoCoverageDriverAvailableExceptionSpec.php [hunk 1]
```diff
@@ -14,7 +15,7 @@ class NoCoverageDriverAvailableExceptionSpec extends ObjectBehavior
 {
     public function it_is_initializable(): void
     {
-        $this->shouldBeAnInstanceOf(\RuntimeException::class);
+        $this->shouldBeAnInstanceOf(RuntimeException::class);
```
Comment (jaylinski) [inline spec/Exception/NoCoverageDriverAvailableExceptionSpec.php]: I'd rather leave this as is and not use an extra `use` statement. Just because it is less code.

File: src/CodeCoverageExtension.php [hunk 2]
```diff
@@ -12,18 +10,24 @@
...
 use PhpSpec\ServiceContainer;
+use RuntimeException;
 use SebastianBergmann\CodeCoverage\CodeCoverage;
 use SebastianBergmann\CodeCoverage\Filter;
 use SebastianBergmann\CodeCoverage\Report;
 use SebastianBergmann\CodeCoverage\Version;
 use Symfony\Component\Console\Input\InputOption;
 
+use function count;
+use function is_array;
```
Comment (jaylinski) [inline src/CodeCoverageExtension.php]: Why do we need to explicitly import those functions? Are they overwritten somewhere? :thinking:

File: src/CodeCoverageExtension.php [hunk 3]
```diff
@@ -12,18 +10,24 @@
  * that was distributed with this source code.
  */
 
+declare(strict_types=1);
```
Comment (jaylinski) [inline src/CodeCoverageExtension.php]: Can you elaborate why you moved that statement?

File: composer.json [hunk 4]
```diff
@@ -39,7 +39,7 @@
     ],
     "require": {
         "php": "^7.1",
-        "phpspec/phpspec": "^4.2 || ^5.0",
+        "phpspec/phpspec": "^4.2 || ^5.0 || ^6.0",
```
Comment (jaylinski) [inline composer.json]: phpspec 6 requires at least PHP 7.2.

Maybe we should release a new major version (with phpspec v6 support) and also update our requirement to `"php": "^7.2"`. :thinking:

**Comment by jaylinski** [comment]
@ptondereau Thanks for PR! I added some comments to your proposed changes.

File: composer.json (hunk 4 above)
Comment (ptondereau) [reply composer.json]: Yes I knew that phpspec require 7.2 but for BC break I still require 7.1

File: src/CodeCoverageExtension.php (hunk 3 above)
Comment (ptondereau) [reply src/CodeCoverageExtension.php]: PHPCsFixer from GrumpPHP does this.

File: src/CodeCoverageExtension.php (hunk 2 above)
Comment (ptondereau) [reply src/CodeCoverageExtension.php]: PHPCsFixer from GrumpPHP does this.

File: spec/Exception/NoCoverageDriverAvailableExceptionSpec.php (hunk 1 above)
Comment (ptondereau) [reply spec/Exception/NoCoverageDriverAvailableExceptionSpec.php]: PHPCsFixer from GrumpPHP does this.

**Comment by ptondereau** [comment]
PHPCsFixer's rules missmatch with Phpcs. I think ruling should be done in another  PR or use an external services for Codestyle such as https://styleci.io/

**Comment by jaylinski** [comment]
@drupol Can you take a look at the grumphp issue?

**Comment by ptondereau** [comment]
@jaylinski grumphp issues are about codestyle. The problem is if I fix PHPCS errors it will throw an error on PHPCSFIXER and vice-versa.

**Comment by drupol** [comment]
Hi, i'll fix this today.

**Comment by ptondereau** [comment]
thank you @drupol! If I fix all files according to the CI, I end up with:
```
PhpCs:
FILE: src/CodeCoverageExtension.php
----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 56 | WARNING | Line exceeds 120 characters; contains 125 characters
----------------------------------------------------------------------
```

But PhpCsFixer don't allow redefining this line

**Comment by drupol** [comment]
Dear @ptondereau ,

I've updated my package, run composer update and normally it should be fixed.

Let me know how it goes, I'm available the whole evening.

**Comment by ptondereau** [comment]
@drupol still complaining about the above error

**Comment by drupol** [comment]
Checking it out myself.

**Comment by drupol** [comment]
Here's the patch that fix your code.
[fix-codestyle.patch.txt](https://github.com/friends-of-phpspec/phpspec-code-coverage/files/3825781/fix-codestyle.patch.txt)

**Comment by ptondereau** [comment]
Thank you I will test IT later

**Comment by ptondereau** [comment]
all good! Thank you very much @drupol!

**Comment by drupol** [comment]
You're welcome :-)

File: src/CodeCoverageExtension.php [hunk 5]
```diff
@@ -64,15 +68,13 @@ public function load(ServiceContainer $container, array $params = []): void
...
+            } elseif (!is_array($options['format'])) {
                 $options['format'] = (array) $options['format'];
             }
 
-            if (isset($options['output'])) {
-                if (!\is_array($options['output']) && 1 === \count($options['format'])) {
-                    $format = $options['format'][0];
-                    $options['output'] = [$format => $options['output']];
-                }
+            if (isset($options['output']) && !is_array($options['output']) && 1 === count($options['format'])) {
```
Comment (drupol) [inline src/CodeCoverageExtension.php]: This is an unrelated changes and should be reverted.

Comment (ptondereau) [reply src/CodeCoverageExtension.php]: Sorry, it's a miss while I was struggling with fixing test. I agree with atomic commit

**Comment by ptondereau** [comment]
@drupol fixed.

**Comment by drupol** [comment]
> PHPCsFixer's rules missmatch with Phpcs. I think ruling should be done in another PR or use an external services for Codestyle such as https://styleci.io/

styleci.io is nice, but you will never reach the customization level that drupol/php-conventions can offer.

**Comment by ptondereau** [comment]
> > PHPCsFixer's rules missmatch with Phpcs. I think ruling should be done in another PR or use an external services for Codestyle such as [styleci.io](https://styleci.io/)
> 
> styleci.io is nice, but you will never reach the customization level that drupol/php-conventions can offer.

It was just a suggestion

File: composer.json (hunk 4 above)
Comment (shulard) [reply composer.json]: Hello !

It's not really a problem here since we are just "allowing" the usage of phpspec 6.0.

Since we are using `^7.1` for the php version constraint, it allow using PHP7.2 without any problem. Then depending on the environment and the project context it will be the composer resolution algorithm that must trigger errors or warnings.

**Comment by ptondereau** [comment]
@shulard any release scheduled?

**Comment by shulard** [comment]
Yep, I've just created the v4.3.2 tag :smile:.

**Comment by drupol** [comment]
Merci :heart:
